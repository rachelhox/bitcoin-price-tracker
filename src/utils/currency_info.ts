interface IStringIndex extends Record<string, any> {}

/**
 * Takes the key name and returns the location. Will return null if undefined
 */

const currencies: IStringIndex = {
  usd: {
    name: 'US Dollar',
    key: 'usd',
    unit: '$',
    type: 'fiat',
  },
  aed: {
    name: 'United Arab Emirates Dirham',
    key: 'aed',
    unit: 'DH',
    type: 'fiat',
  },
  eur: {
    name: 'Euro',
    key: 'eur',
    unit: '€',
    type: 'fiat',
  },
  aud: {
    name: 'Australian Dollar',
    key: 'aud',
    unit: 'A$',
    type: 'fiat',
  },
  cad: {
    name: 'Canadian Dollar',
    key: 'cad',
    unit: 'CA$',
    type: 'fiat',
  },
  chf: {
    name: 'Swiss Franc',
    key: 'chf',
    unit: 'Fr.',
    type: 'fiat',
  },
  gbp: {
    name: 'British Pound Sterling',
    key: 'gbp',
    unit: '£',
    type: 'fiat',
  },
  hkd: {
    name: 'Hong Kong Dollar',
    key: 'hkd',
    unit: 'HK$',
    type: 'fiat',
  },
  sgd: {
    name: 'Singapore Dollar',
    key: 'sgd',
    unit: 'S$',
    type: 'fiat',
  },
  cny: {
    name: 'Chinese Yuan',
    key: 'cny',
    unit: '¥',
    type: 'fiat',
  },
  jpy: {
    name: 'Japanese Yen',
    key: 'jpy',
    unit: '¥',
    type: 'fiat',
  },
  inr: {
    name: 'Indian Rupee',
    key: 'inr',
    unit: '₹',
    type: 'fiat',
  },
  krw: {
    name: 'South Korean Won',
    key: 'krw',
    unit: '₩',
    type: 'fiat',
  },
};

export const getCurrencyInfo = (key: string | number) => {
  return currencies[key] ?? {};
};
