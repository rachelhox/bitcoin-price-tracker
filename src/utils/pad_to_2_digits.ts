/* eslint-disable */

export const padTo2Digits = (num: any) => {
  return String(num).padStart(2, '0');
};
