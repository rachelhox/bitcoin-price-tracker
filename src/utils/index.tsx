import { getCurrencyInfo } from './currency_info';
import { convertToMoney } from './convert_to_money';
import { padTo2Digits } from './pad_to_2_digits';

export { getCurrencyInfo, convertToMoney, padTo2Digits };
