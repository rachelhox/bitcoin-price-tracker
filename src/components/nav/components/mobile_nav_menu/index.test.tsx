/* eslint-disable react/jsx-curly-brace-presence */
import renderer from 'react-test-renderer';
import { MockTheme, wait } from '@tests/utils';
import MobileNavMenu from '.';

// ==================================
// global setup
// ==================================
let component: renderer.ReactTestRenderer;

// ==================================
// mocks
// ==================================
jest.mock('..', () => ({
  CurrencyMenuButton: (props: any) => (
    <div id="currencyMenuButton" {...props} />
  ),
}));

// ==================================
// unit tests
// ==================================
describe('screen: Nav/Mobile Nav Menu', () => {
  beforeEach(async () => {
    component = renderer.create(
      <MockTheme>
        <MobileNavMenu link="/" />
      </MockTheme>
    );
    await wait();
  });

  it('it renders', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

afterEach(() => {
  jest.clearAllMocks();
});
