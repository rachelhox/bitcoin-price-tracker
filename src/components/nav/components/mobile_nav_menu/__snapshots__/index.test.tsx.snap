// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: Nav/Mobile Nav Menu it renders 1`] = `
<div>
  <button
    aria-haspopup="true"
    aria-label="more"
    className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-penbf1-MuiButtonBase-root-MuiIconButton-root"
    disabled={false}
    id="long-button"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <svg
      fill="none"
      height="18"
      viewBox="0 0 24 18"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1 9L23 9M1 1L23 1M1 17L23 17"
        stroke="#FFFFFF"
        strokeLinecap="round"
        strokeWidth="2"
      />
    </svg>
    <span
      className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
